discovery_engine/lib/src/ffi/types/weighted_source_vec.dart

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import 'dart:ffi';
+
+import 'package:xayn_discovery_engine/src/domain/models/source_reacted.dart';
+import 'package:xayn_discovery_engine/src/ffi/genesis.ffigen.dart';
+import 'package:xayn_discovery_engine/src/ffi/load_lib.dart' show ffi;
+import 'package:xayn_discovery_engine/src/ffi/types/box.dart';
+import 'package:xayn_discovery_engine/src/ffi/types/list.dart';
+import 'package:xayn_discovery_engine/src/ffi/types/weighted_source.dart';
+
+final _adapter = ListFfiAdapter(
+  alloc: ffi.alloc_uninitialized_weighted_source_slice,
+  next: ffi.next_weighted_source,
+  writeNative: (source, place) => source.writeNative(place),
+  readNative: WeightedSourceFfi.readNative,
+  getVecLen: ffi.get_weighted_source_vec_len,
+  getVecBuffer: ffi.get_weighted_source_vec_buffer,
+  writeNativeVec: ffi.init_weighted_source_vec_at,
+);
+
+extension WeightedSourceListFfi on List<SourceReacted> {
+  static List<WeightedSourceFfi> readSlice(
+    final Pointer<RustWeightedSource> ptr,
+    final int len,
+  ) =>
+      _adapter.readSlice(ptr, len);
+
+  Boxed<RustVecWeightedSource> allocBoxedVec() {
```

Comment:
🔧 unfortunately the ffi naming is not consistent as this is sometimes named `allocVec` and sometimes `allocNative`, but the box is usually left out (i know that not even this is done everywhere, eg for the `consume_` methods, imo the whole ffi should be named consistently, but i guess it's not worth the effort anymore if we move to protomessages anyways :/)
```suggestion
  Boxed<RustVecWeightedSource> allocVec() {
```

Reply:
good to nevertheless point out - renamed this and also `consumeBoxedVec`